controllers/controller_ref_manager.go

```diff
@@ -551,3 +523,247 @@ func CheckStatefulSet(etcd *druidv1alpha1.Etcd, statefulSet *appsv1.StatefulSet)
 
 	return nil
 }
+
+func getMapFromEtcd(im imagevector.ImageVector, etcd *druidv1alpha1.Etcd) (map[string]interface{}, error) {
```

Comment:
Why has this function been moved to this file? As far as I can see, it's still used only in `etcd_controller.go` and so it belongs to that file (or to its own file, or even its own type, something like e.g. `ValuesProvider`), but definitely not here?